scripts/sync-kernel.sh

```diff
@@ -0,0 +1,96 @@
+SQUASHED_BASELINE_COMMIT=$(git commit-tree ${BASELINE_COMMIT}^{tree} -m "BASELINE ${BASELINE_COMMIT}")
+echo "SQUASHED BASELINE COMMIT: ${SQUASHED_BASELINE_COMMIT}"
+
+# Re-apply the rest of commits
+git rebase --onto ${SQUASHED_BASELINE_COMMIT} ${BASELINE_COMMIT} libbpf-${SUFFIX}
+
+# Move all libbpf files into libbpf-projection directory
+git filter-branch --prune-empty -f --tree-filter '                                                                             \
+    mkdir -p libbpf-projection/include/uapi/linux libbpf-projection/include/tools &&                                           \
+    git mv -kf tools/lib/bpf libbpf-projection/src &&                                                                          \
+    git mv -kf tools/include/uapi/linux/{bpf_common.h,bpf.h,btf.h,if_link.h,netlink.h} libbpf-projection/include/uapi/linux && \
+    git mv -kf tools/include/tools/libc_compat.h libbpf-projection/include/tools &&                                            \
+    git rm --ignore-unmatch -f libbpf-projection/src/{Makefile,Build,test_libbpf.cpp,.gitignore}                               \
+  ' libbpf-${SUFFIX}
+
+# Make libbpf-projection a new root directory
+git filter-branch --prune-empty -f --subdirectory-filter libbpf-projection libbpf-${SUFFIX}
+
+# We don't want to include SQUASHED_BASELINE_COMMIT into patch set,
+# but git doesn't allow to exclude rev A in A..B rev range, so instead
+# we calculate number of patches to include
+COMMIT_CNT=$(git rev-list --count ${SQUASHED_BASELINE_COMMIT}..libbpf-${SUFFIX})
+
+# If baseline is the only commit with libbpf-related changes, bail out
+if ((${COMMIT_CNT} <= 1)); then
+    echo "No new changes to apply found, we are done!"
+    exit 2
+fi
+
+# Exclude baseline commit and generate nice cover letter with summary
+git format-patch -$((${COMMIT_CNT}-1)) --cover-letter -o /tmp/libbpf-${SUFFIX}.patches
+
+# Use generated cover-letter as a template for "sync commit" with
+# baseline and checkpoint commits from kernel repo (and leave summary
+# from cover letter intact, of course)
+echo ${CHECKPOINT_COMMIT} > CHECKPOINT-COMMIT &&                                                 \
+git add CHECKPOINT-COMMIT &&                                                                     \
+awk '/\*\*\* BLURB HERE \*\*\*/ {p=1} p' /tmp/libbpf-${SUFFIX}.patches/0000-cover-letter.patch | \
+sed "s/\*\*\* BLURB HERE \*\*\*/\
+sync: latest libbpf changes from kernel\n\
+\n\
+Syncing latest libbpf commits from kernel repository.\n\
+Baseline commit:   ${BASELINE_COMMIT}\n\
+Checkpoint commit: ${CHECKPOINT_COMMIT}/" |                                                      \
+git commit --file=-
+
+# Now generate single-file patchset to apply on top of libbpf repo,
+# this time including "sync commit" as well
+git format-patch -${COMMIT_CNT} --stdout > /tmp/libbpf-${SUFFIX}.patchset
+echo "PATCHSET: /tmp/libbpf-${SUFFIX}.patchset"
+
+cd ${LIBBPF_REPO}
+git co -b libbpf-sync-${SUFFIX}
```

Comment:
I assume this is `git checkout`?

Reply:
Ah, yes, it's my local alias. I'm so used to it that didn't even realize it's not standard feature, will fix.